app/src/main/java/server/ServerResponse.java

```diff
@@ -19,16 +19,48 @@ public ServerResponse(ServerRequest request, HashMap<String, Route> routeList) {
     private String returnResponse(ServerRequest requestObject){
         String routeVersion = String.valueOf(requestObject.requestLine.get("HTTPVersion"));
         String routePath = String.valueOf(requestObject.requestLine.get("Path"));
+        String routeMethod = String.valueOf(requestObject.requestLine.get("Method"));
 
-        Route obj = this.routeList.get(routePath);
+        response = this.assembleResponse(routeVersion, routePath, routeMethod);
+
+        return response;
+
+    }
+
+    private String assembleResponse(String routeVersion, String routePath, String routeMethod){
+        Route obj = this.routeMap.get(routePath);
 
         if (obj != null){
-            return obj.getObjectResponse(routeVersion, this.getHeaders(), this.currentRequest.body);
+            if (!checkIfMethodIsAllowed(this.currentRequest).equals("")){
+                return checkIfMethodIsAllowed(this.currentRequest);
+            } else{
+                String response = obj.getObjectResponse(routeVersion, this.getHeaders(), this.currentRequest.body);
+                response = trimBody(response, routePath, routeMethod);
```

Comment:
Would it be simpler to check whether the client is asking for HEAD or OPTIONS before calling `getObjectResponse`, similar to the check for methodNotAllowed? Seems that in those cases you don't use any of the data that `getObjectResponse` returns.

Reply:
Since the route is predetermined by the "/address" it actually does use the getObjectResponse instead of a build generic. ie some of the responses don't contain the header list while others do. To keep it this way it might still just be easier to trim the response than a preemptive check.

Reply:
Talked IRL